src/ZPin.cpp

```diff
@@ -578,29 +560,22 @@ int ZPin::enableRiseFallEvents(int eventType)
         if (!(status & IO_STATUS_DIGITAL_IN))
             getDigitalValue();
 
-        enable_irqs();
+        const mcu_pin_obj_t* pin = samd_peripherals_get_pin(name);
+        CODAL_ASSERT(pin != NULL);
+        CODAL_ASSERT(pin->has_extint);
 
-        int pin = (int)name & PINMASK;
+        EICFactory eic;
+        this->chan = eic.getInstance(pin->extint_channel);
+        CODAL_ASSERT(chan != NULL);
 
-        eventPin[pin] = this;
-
-        volatile uint32_t *ptr = &SYSCFG->EXTICR[pin >> 2];
-        int shift = (pin & 3) * 4;
-        int port = (int)name >> 4;
-
-        // take over line for ourselves
-        *ptr = (*ptr & ~(0xf << shift)) | (port << shift);
-
-        EXTI->EMR &= ~GPIO_PIN();
-        EXTI->IMR |= GPIO_PIN();
-        EXTI->RTSR |= GPIO_PIN();
-        EXTI->FTSR |= GPIO_PIN();
+        gpio_set_pin_pull_mode(name, map(pullMode));
```

Comment:
I'll remove these, I guess not really required.

Reply:
This was left-over from STM